### PR TITLE
[search] Fixed search in cian mode.

### PR DESCRIPTION
**Pull request by ygorshenin**

### CODE REVIEW

**Comment by ygorshenin** [comment]
https://jira.mail.ru/browse/MAPSME-5192

File: map/framework.cpp [hunk 1]
```diff
@@ -1392,6 +1392,8 @@ bool Framework::SearchInViewport(search::ViewportSearchParams const & params)
     strings::AsciiToLower(query);
     if (query == "cian")
       m_cianSearchMode = true;
```
Comment (rokuz) [inline map/framework.cpp]: m_cianSearchMode = (query == "cian");

File: search/geocoder.cpp [hunk 2]
```diff
@@ -343,6 +339,10 @@ Geocoder::~Geocoder() {}
 void Geocoder::SetParams(Params const & params)
 {
   m_params = params;
+  if (m_params.m_cianMode)
```
Comment (rokuz) [inline search/geocoder.cpp]: m_model.SetCianEnabled(m_params.m_cianMode);

File: search/geocoder.hpp [hunk 3]
```diff
@@ -233,7 +232,7 @@ class Geocoder
 
   // This field is used to map features to a limited number of search
   // classes.
-  SearchModel const & m_model;
+  SearchModel m_model;
```
Comment (rokuz) [inline search/geocoder.hpp]: Тут не по ошибке ссылка удаляется?

File: search/model.hpp [hunk 4]
```diff
@@ -47,16 +47,14 @@ class SearchModel
     SEARCH_TYPE_COUNT
   };
 
-  static SearchModel const & Instance();
-
   SearchType GetSearchType(FeatureType const & feature) const;
 
-private:
-  SearchModel() = default;
+  void EnableCian() { m_cianMode = true; }
```
Comment (rokuz) [inline search/model.hpp]: void SetCianEnabled(bool enabled) { m_cianMode = enabled; }

File: search/geocoder.cpp (hunk 2 above)
Comment (ygorshenin) [inline search/geocoder.cpp]: Not sure that it's more readable, but surely more compact. Done.

File: search/geocoder.hpp (hunk 3 above)
Comment (ygorshenin) [inline search/geocoder.hpp]: No, everything is fine. SearchModel is stateful now, so Geocoder should have its own instance of SearchModel.

Comment (rokuz) [inline search/geocoder.hpp]: ok

File: map/framework.cpp (hunk 1 above)
Comment (ygorshenin) [inline map/framework.cpp]: Done

File: search/model.hpp (hunk 4 above)
Comment (ygorshenin) [inline search/model.hpp]: Done

**Comment by ygorshenin** [comment]
@rokuz PTAL

**Pull request merged**